File: python/fetch_recipes.py
import requests
import json

query = "pizza"

n = 0
m = 99

url = f"https://www.edamam.com/search?q=pizza&health=vegan&from={n}&to={m}"

data = requests.get(url)

results = data.json()
count = results["count"]

with open(f'recipes/{query}_{n}_to_{m}.json', "w", encoding="utf-8") as f:
  json.dump(results, f, ensure_ascii=False, indent=4)

if count > 99:
  n = n + 100
  m = m + 100
  while n <= count:
    extra_data = requests.get(f"https://www.edamam.com/search?q=pizza&health=vegan&from={n}&to={m}").json()
    with open(f'recipes/{query}_{n}_to_{m}.json', "w", encoding="utf-8") as f:
      json.dump(extra_data, f, ensure_ascii=False, indent=4)
    n = n + 100
    m = m + 100

File: static/fetch_recipes.rb
require 'rest-client'
require 'json'

path = "recipes"
query = "burger"

m = 0
n = 99

url = "https://www.edamam.com/search?q=#{query}&health=vegan&from=#{m}&to=#{n}"
data = RestClient.get(url)
count = JSON.parse(data)["count"]

puts "No. of recipes: #{count}"

File.open(File.join(__dir__, path, "#{query}_#{m}_to_#{n}.json"), "w+") do |f|
  f.write(data)
  puts "Saved file #{f.path}"
end

if count > 99
  m += 100
  n += 100
  while m < count
    url = "https://www.edamam.com/search?q=#{query}&health=vegan&from=#{m}&to=#{n}"
    data = RestClient.get(url)
    File.open(File.join(__dir__, path, "#{query}_#{m}_to_#{n}.json"), "w+") do |f|
      f.write(data)
      puts "Saved file #{f.path}"
    end
    m += 100
    n += 100
  end
end
